### ISSUE TITLE
Keydown event doesn't fire properly

### ISSUE THREAD

**Issue by ghost** [wexond/browser-base#518]
**Bug description**
Hi everyone, 

I'll try to keep this short. Maybe I'm not getting this right, but I'm trying to add "keydown (enter)" event on the address bar. Now, it's already added in the "AddressBar -> index.tsx" file, but for the first time when you open the browser, click on address bar, type a URL and press enter, it doesn't fire. 

But after doing it once, when you click on address bar again, re-type and press enter, it fires. It's probably related to state management in some way, but I'm just wanting somebody to help me understand what might be the issue. 

The requirement I want (custom for myself), is that if somebody writes "facebook.com" and tries to visit (e.g. through pressing enter key), then they shouldn't be able to do that and page should redirect instead to "google.com" 

For the second time when I handle logic in "keydown" event handler, it works but not for the first time. 

Any help would be appreciated, thanks.

**Comment by sentialx** [comment]
The omnibox popup is located in a different view out of the DOM.

**Comment by ghost** [comment]
Thanks for replying! So, how can I achieve this functionality? Can you point me in the right direction please?

**Comment by sentialx** [comment]
Just make your keydown event in the omnibox popup, not the address bar.

**Comment by ghost** [comment]
Sir, I'm a little new to React and that's why I was confused. Please tell me where can I find this omnibox file, is it something like "index.tsx" file for address bar component ? Please a little more help is required, thank you

**Comment by sentialx** [comment]
https://github.com/wexond/desktop/blob/master/src/renderer/views/search/components/App/index.tsx

What's your use case btw?

**Comment by ghost** [comment]
Sorry for replying late. I tried handling logic in the "keypress" event in the file you had shared, it works! Thank you very much dear sir!

First, let me explain the use case: 

//This functionality is achieved
User clicks on address bar -> types "facebook.com" -> it should redirect instead to "google.com". 

//This functionality remains to be achieved
When user gets redirected to "google.com", even though page loaded is google, the address bar should still say "facebook.com". 

Basically this custom check is only for facebook for now. 

Now, how do I change the URL bar address only and NOT reload the page you know? Can you please guide me about that?

Thanks
